apps/wallet-api-tools/src/components/Message.tsx

```diff
@@ -0,0 +1,40 @@
+import React, { CSSProperties, Suspense } from "react";
+import CodeMirror, { Extension } from "@uiw/react-codemirror";
+import { langs } from "@uiw/codemirror-extensions-langs";
+import { MessageIn, MessageOut } from "../types";
+
+type Props = {
+  message: MessageIn | MessageOut;
+  style?: CSSProperties;
+  theme?: Extension;
+  date?: string;
+};
+
+const Message = ({ message, theme, date }: Props) => {
+  return (
+    <>
+      <p className="text-slate-400 text-sm mb-2">{date}</p>
+      <Suspense fallback={"..."}>
+        <CodeMirror
+          value={JSON.stringify(message.value, null, 2)}
```

Comment:
It might be good to memo the `JSON.stringify` at some point and after doing some tests with the react devtools to look at the re-rendering cost on bigger list on different actions

Reply:
What about using a memo on the entire component [Memo](https://react.dev/reference/react/memo) so it render once